### ISSUE TITLE
[octavia-ingress-controller] Doesn't publish the full certificate chain

### ISSUE THREAD

**Issue by dr4Ke** [kubernetes/cloud-provider-openstack#1220]
**Is this a BUG REPORT or FEATURE REQUEST?**:

To me, this is a bug. But it could also be viewed as a feature request.

> Uncomment only one, leave it on its own line: 
>
/kind bug
> /kind feature

**What happened**:
The ingress controller created a `TERMINATED_HTTPS` load balancer using a provided TLS secret. The LB is responding to requests presenting only the leaf certificate, not the chain.

**What you expected to happen**:
The LB should present the leaf certificate **and** any intermediate certificates needed (fullchain)

**How to reproduce it**:
- Create a TLS secret in kubernetes, with the full chain in `tls.crt`
```
kubectl create secret tls my-cert --cert cert.pem --key key.pem
```
- Create an octavia ingress
```yaml
cat <<EOF | kubectl apply -f -
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  annotations:
     kubernetes.io/ingress.class: openstack
  name: nginx
spec:
  rules:
  - host: host.domain.com
    http:
      paths:
      - backend:
          serviceName: nginx-http
          servicePort: 80
  tls:
  - hosts:
    - host.domain.com
    secretName: my-cert
EOF
```
- Inspect the presented certificates:
```
openssl s_client -connect host.domain.com:443
[...]
---
Certificate chain
 0 s:CN = host.domain.com
   i:C = US, O = Let's Encrypt, CN = Let's Encrypt Authority X3
---
```

A manually configured barbican TLS secret, with full chain show something like:
```
---
Certificate chain
 0 s:CN = host.domain.com
   i:C = US, O = Let's Encrypt, CN = Let's Encrypt Authority X3
 1 s:C = US, O = Let's Encrypt, CN = Let's Encrypt Authority X3
   i:O = Digital Signature Trust Co., CN = DST Root CA X3
---
```

**Anything else we need to know?**:

**Environment**:
- openstack-cloud-controller-manager(or other related binary) version: 1.19.0
- OpenStack version: Train
- Others:

**Comment by lingxiankong** [comment]
Hi @dr4Ke , may I know what do you mean by `A manually configured barbican TLS secret`?

**Comment by dr4Ke** [comment]
I meant using the openstack cli using this scenario for example:
https://docs.openstack.org/octavia/latest/user/guides/basic-cookbook.html#deploy-a-tls-terminated-https-load-balancer

```
openssl pkcs12 -export -inkey server.key -in server.crt -certfile ca-chain.crt -passout pass: -out server.p12
openstack secret store --name='tls_secret1' -t 'application/octet-stream' -e 'base64' --payload="$(base64 < server.p12)"
openstack loadbalancer create --name lb1 --vip-subnet-id public-subnet
# Re-run the following until lb1 shows ACTIVE and ONLINE statuses:
openstack loadbalancer show lb1
openstack loadbalancer listener create --protocol-port 443 --protocol TERMINATED_HTTPS --name listener1 --default-tls-container=$(openstack secret list | awk '/ tls_secret1 / {print $2}') lb1
openstack loadbalancer pool create --name pool1 --lb-algorithm ROUND_ROBIN --listener listener1 --protocol HTTP
openstack loadbalancer member create --subnet-id private-subnet --address 192.0.2.10 --protocol-port 80 pool1
openstack loadbalancer member create --subnet-id private-subnet --address 192.0.2.11 --protocol-port 80 pool1
```

**Comment by lingxiankong** [comment]
In this command

> kubectl create secret tls my-cert --cert cert.pem --key key.pem

How did you generate the cert.pem and key.pem?

**Comment by dr4Ke** [comment]
The key was generated with openssl. The cert.pem is the "fullchain" given by lets encrypt after a successful challenge.

I will check again everything, and provide more precise information.

**Comment by birnemeier** [comment]
Hi, I ran into the same issue. 

When cert.pem contains intermediate certificates these intermediate certificates are not copied over to Barbican store. This is not a bug it is marked as TODO by @lingxiankong here:

https://github.com/kubernetes/cloud-provider-openstack/blob/4254f067576aa88a3287c00195d28d58a755a681/pkg/ingress/controller/controller.go#L609

The full PEM file is parsed, by only the first cert is copied over.

**Comment by klippo** [comment]
We just ran into the same issue. Are there any plans to support this in the near future?

**Comment by lingxiankong** [comment]
Hi guys, I agree this is a bug that needs to be fixed, but I'm not available for a PR at the moment, could any one of you pick up this issue?

**Comment by hedinasr** [comment]
resolve by #1699 ?

**Comment by lingxiankong** [comment]
> resolved by #1699 ?

Yes, should be. @dr4Ke I am closing this issue, please re-open if you think it's still valid.
